ddl/table.go

```diff
@@ -134,12 +139,17 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 
 	// just decode, createTable will use it as Args[0]
 	tbInfo := &model.TableInfo{}
-	if err := job.DecodeArgs(tbInfo); err != nil {
+	fkCheck := false
+	if err := job.DecodeArgs(tbInfo, &fkCheck); err != nil {
 		// Invalid arguments, cancel this job.
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
 
+	if len(tbInfo.ForeignKeys) > 0 {
+		return createTableWithForeignKeys(d, t, job, tbInfo, fkCheck)
```

Comment:
I found `createTableWithForeignKeys` will also call `checkTableForeignKeyValidInOwner` to check the tables referred to it, will this check be skipped if a new created table has no foreign keys but referred by other tables?

Reply:
Well, this is a double-check, check here uses to return as early as possible if the table is invalid, and reduce ddl job count.

Reply:
What I mean is that currently `onCreateTable` will not check the tables referred to it, for example here are two ddl:

create table t1(x int primary key);
create table t2(a int, FOREIGN KEY (a) REFERENCES t1(id)); 

1. set @@foreign_key_checks=0;
2. t1 is checked but not submitted to ddl owner yet.
3. t2 is checked but not submitted to ddl owner yet.
4. execute create t2 in ddl owner.
5. execute create t1 in ddl onwer.

t1 should fail because it does not has a column 'id'. But I think it will be success for the current implement.

Reply:
`checkTableForeignKeyValidInOwner` function has also checked the referred foreign key info now:

<img width="1218" alt="image" src="https://user-images.githubusercontent.com/26020263/185847248-9ffb99b1-0a9b-4afa-88ca-ef342ca9378e.png">